terraform/modules/network-validator/scripts/network_validation.sh

```diff
@@ -250,14 +258,18 @@ function check_tcp_port_open_in_seclist_or_nsg() {
 
       if [[ ($ingress_protocol = "all" || $ingress_protocol = $tcp_protocol ) && ( $tcp_options = "null" || ( $port -ge $port_min && $port -le $port_max ) ) && $source_in_cidr_range -eq 0 ]]
       then
-         port_is_open=true
-         echo 0
-         return
+        port_is_open=true
+        echo 0
+        return
       fi
     done
   fi
-
-  echo 1
+  if [[ ${#nsg_sec_list_array[@]} != 0 ]]
+  then
+    echo "${nsg_sec_list_array[@]}"
+  else
```

Comment:
If the first ingress rule has NSG as a source - then the warning array is populated (echo warning).
If the second ingress rule has a CIDR block as a source, and the port is not open (not in the CIDR range) - then it's an error condition (echo 1).

However, won't this block only print a warning, and the error condition is missed?

Is this case tested?

Reply:
I agree that this is very ugly, but we need a quick fix. My comments on why I think this will "solve" this issue for now:
case 1 - nsg with port open to CIDR range. Error 0 returned.
case 2 - all nsg's use CIDR ranges, but none are the range we need. Error 1 returned.
case 3 - some nsg's reference other nsgs, and some reference CIDR ranges, but none that reference CIDR ranges are the range we need. Warning returned, because it is indeterminate unless we recursively search the CIDR references until we get to a CIDR. We will have a task in the backlog to do this in the future.